hphp/runtime/vm/jit/vasm-arm.cpp

```diff
@@ -536,9 +536,15 @@ void Vgen::emit(const phplogue& i) {
 }
 
 void Vgen::emit(const phpret& i) {
-  a->Ldr(X(rlr()), X(i.fp)[AROFF(m_savedRip)]);
-  if (!i.noframe) {
-    a->Ldr(X(i.d), X(i.fp)[AROFF(m_sfp)]);
+  const int sizeofXregOnStack = 8;
+  // prefer load-pair instruction
+  if (!i.noframe && AROFF(m_savedRip) == sizeofXregOnStack && AROFF(m_sfp) == 0) {
```

Comment:
I don't think the offset checks are necessary.  As we're in the ARM vasm-to-machine-code lowering state, the dependency on the layout of the frame is already immediately apparent.

lgtm otherwise though.